src/addrman.h

```diff
@@ -304,8 +336,14 @@ class CAddrMan
 
         unsigned char nVersion = 1;
         s << nVersion;
-        s << ((unsigned char)32);
+        s << ((unsigned char)36);
         s << nKey;
+        unsigned long long asmap_version = 0;
+        if (m_asmap.size() != 0) {
+            std::hash<std::vector<bool>> hasher;
+            asmap_version = hasher(m_asmap);
```

Comment:
I don't think using std::hash is the right choice here, as it has no well-defined semantics (even updating your c++ stdlib could change the function). You can use `SerializeHash(m_asmap)` instead.